web/src/drafts.js

```diff
@@ -318,13 +318,6 @@ export function format_draft(draft) {
         let sub;
         if (draft.stream_id) {
             sub = sub_store.get(draft.stream_id);
-        } else if (draft.stream && draft.stream.length > 0) {
-            // draft.stream is deprecated but might still exist on old drafts
-            stream_name = draft.stream;
-            const sub = stream_data.get_sub(stream_name);
-            if (sub) {
-                draft.stream_id = sub.stream_id;
-            }
```

Comment:
Given how updates work, with not everyone running this code for 30 days, is this actually a safe change? If not, what would be a good way to handle this with typescript?

Reply:
It's not safe to delete this; we'll need to keep some version of that code for another couple years. 

A possible answer for both this and https://github.com/zulip/zulip/pull/28731#discussion_r1480939132 is for the `zod` parsing code to be responsible for updating the draft format when reading it out of local storage.

Reply:
This has been addressed by using `zod`

Reply:
The commit message for deleting suggests this code was not operative, which is wrong -- self-hosted systems can upgrade on arbitrary schedules.

Reply:
Updated the commit message.

Reply:
Great point, rearranged the commits to do that zod work first